lib/discordrb/data/activity.rb

```diff
@@ -0,0 +1,235 @@
+    LISTENING = 2
+    # This type is currently unused in the client but can be reported by bots
+    WATCHING = 3
+    # Type indicating the activity is a custom status
+    CUSTOM = 4
+
+    # @!visibility private
+    def initialize(data, bot)
+      @name = data['name']
+      @type = data['type']
+      @url = data['url']
+      @application_id = data['application_id']
+      @details = data['details']
+      @state = data['state']
+      @instance = data['instance']
+      @flags = data['flags'] || 0
+
+      @timestamps = Timestamps.new(data['timestamps']) if data['timestamps']
+      @secrets = Secret.new(data['secrets']) if data['secrets']
+      @assets = Assets.new(data['assets']) if data['assets']
+      @party = Party.new(data['party']) if data['party']
+      # Merge of #671 requires this to be changed.
+      @emoji = Emoji.new(data['emoji'], bot, nil) if data['emoji']
+    end
+
+    # @return [true, false] Whether or not the `join` flag is set for this activity
+    def join?
+      flag_set? :join
+    end
+
+    # @return [true, false] Whether or not the `spectate` flag is set for this activity
+    def spectate?
+      flag_set? :spectate
+    end
+
+    # @return [true, false] Whether or not the `join_request` flag is set for this activity
+    def join_request?
+      flag_set? :join_request
+    end
+
+    # @return [true, false] Whether or not the `sync` flag is set for this activity
+    def sync?
+      flag_set? :sync
+    end
+
+    # @return [true, false] Whether or not the `play` flag is set for this activity
+    def play?
+      flag_set? :play
+    end
+
+    # @return [true, false] Whether or not the `instance` flag is set for this activity
+    def instance?
+      @instance || flag_set?(:instance)
+    end
+
+    # @!visibility private
+    def flag_set?(sym)
+      !(@flags & FLAGS[sym]).zero?
+    end
+
+    # Timestamps for the start and end of instanced activities
+    class Timestamps
+      # @return [Time, nil]
+      attr_reader :start
+
+      # @return [Time, nil]
+      attr_reader :end
+
+      # @!visibility private
+      def initialize(data)
+        @start = Time.at(data['start'] / 1000) if data['start']
+        @end = Time.at(data['end'] / 1000) if data['end']
+      end
+    end
+
+    # Contains secrets used for rich presence
+    class Secrets
+      # @return [String, nil] secret for joining a party
+      attr_reader :join
+
+      # @return [String, nil] secret for spectating
+      attr_reader :spectate
+
+      # @return [String, nil] secret for a specific instanced match
+      attr_reader :match
+
+      # @!visibility private
+      def initialize(data)
+        @join = data['join']
+        @spectate = data['spectate']
+        @match = data['match']
+      end
+    end
+
+    # Assets for rich presence images and hover text
+    class Assets
+      # @return [String, nil] the asset id for the large image of this activity
+      attr_reader :large_image
```

Comment:
```suggestion
      attr_reader :large_image_id
```
and provide
```rb
def large_image_url
  # ..
end
```
Per  CDN `app-assets/application_id/asset_id.png`
See how we implement the other CDN resources, i.e. `User#avatar_url` and [API docs](https://discordapp.com/developers/docs/reference#image-formatting-cdn-endpoints)

Ditto for `small_image`